src/transforms/util/mod.rs

```diff
@@ -1,2 +1,9 @@
 #[cfg(any(feature = "transforms-lua"))]
 pub mod runtime_transform;
+
+#[cfg(any(feature = "sources-kubernetes-logs"))]
```

Comment:
Curious to hear @lukesteensen on this one.

These are clearly useful from a composability perspective, and probably can be useful for ideas like future pipelines/architecture/etc. I think enabling them in core right now will give warnings that are out of scope, can you toss like `TODO: Make core vector` on all these?

Reply:
Yes, I also wanted to make them core Vector. I feel we might want to rethink the locations of the files a bit, to make it better suited for such things. We can discuss this. I don't want to do it in this PR though. I'll just add the to do comment. :+1:

Reply:
I'm curious why all of these are being introduced as part of the k8s feature. It seems like there could have been a smaller, simpler implementation that introduced fewer new concepts.

I understand the desire to create new useful building blocks, but I'm skeptical that we've actually done that without discussing the design and how it'd mesh with other use cases. And if the design needs changes to be reusable (usually true) or turns out only to be needed in this one place (very often true), all the extra code and concepts here are purely a maintenace burden that slow us down.

Again, I think our goal here should have been to precisely solve the specific problem at hand. If we then want to extract and reuse portions of that implementation, we can do so when we have sufficient knowledge about the other use cases.